### ISSUE TITLE
optional attributes without values

### ISSUE THREAD

**Issue by nsheff** [epigen/looper#159]
if you have a study with mixed single/paired data, coming from 2 files, then you have a `read2` column, and looper will pass `--input2` with the value in the `read2` column.

But if you have a single-end sample in there, the `read2` column will be blank, and looper chokes. 

PR #157 included an attempt to solve this, but it still fails because the python thinks the value is not `None` or `""`, which are tested for, but `nan`. See this command constructed by looper:

`
/home/nsheff/code/rnapipe/src/rnaKallisto.py  --single-or-paired single --sample-yaml /athena/melnicklab/users/scratch/okbm4/processed/aml_scrna/submission/sample5.yaml --sample-name sample5 --input blah_R1_001.fastq.gz --input2 nan -D -C ...
`
it says: **--input2 nan ** which is still failing...

**Comment by nsheff** [comment]
So, question: why is python reading an empty column in as `nan` instead of as an empty string?

**Comment by vreuter** [comment]
Probably the default behavior of the `pandas` parser. Probably also parameterized, though.

**Comment by nsheff** [comment]
Hmm -- but in looper 0.6 it passes an empty string... so it's something to do with changes in #157.

**Comment by nsheff** [comment]
pandas is indeed reading those blank lines as `NaN`. This is relevant: https://stackoverflow.com/questions/10867028/get-pandas-read-csv-to-read-empty-values-as-empty-string-instead-of-nan

we can use `keep_default_na=False`: 

observe:

```
>>> a = pandas.read_table("metadata/microtest.csv", sep=",")
>>> a["read2"]
0    source2
1    source2
2    source2
3    source2
4        NaN
Name: read2, dtype: object
>>> a.fillna('')["read2"]
0    source2
1    source2
2    source2
3    source2
4           
Name: read2, dtype: object
>>> b = pandas.read_table("metadata/microtest.csv", sep=",", keep_default_na=False)
>>> b["read2"]
0    source2
1    source2
2    source2
3    source2
4           
Name: read2, dtype: object
```

**Comment by nsheff** [comment]
PS this doesn't explain how it switched from `0.6` to `0.7` -- somewhere the `NaN` was getting converted to an empty string in `0.6` I guess. But I think we should read empty strings as empty strings generally, and not as `NaN`s.

**Comment by nsheff** [comment]
Fixed by #160
